app/views/comments/_form.html.erb

```diff
@@ -1,6 +1,8 @@
-<%= form_for [commentable, Comment.new] do |f| %>
-  <div class="form-group">
-    <%= f.text_area :body, class: "form-control", placeholder: "Add a comment" %>
-  </div>
+<% if user_signed_in? %>
+  <%= form_for [commentable, Comment.new] do |f| %>
+    <div class="form-group">
+      <%= f.text_area :body, class: "form-control", placeholder: "Add a comment" %>
+    </div>
   <%= f.submit class: "btn btn-primary" %>
+  <% end %>
 <% end %>
```

Comment:
if user is not logged in, show a message saying "Please login or sign up to comment.". Link login and sign up words to the correct pages.